overlord/hookstate/ctlcmd/ctlcmd.go

```diff
@@ -32,10 +32,19 @@ import (
 	"github.com/snapcore/snapd/strutil"
 )
 
+func createMissingContextError(subcommand string) error {
```

Comment:
(nitpick) I wonder if something like:
```go
index 20b8b09067..76bac68240 100644
--- a/overlord/hookstate/ctlcmd/ctlcmd.go
+++ b/overlord/hookstate/ctlcmd/ctlcmd.go
@@ -32,8 +32,12 @@ import (
        "github.com/snapcore/snapd/strutil"
 )
 
-func createMissingContextError(subcommand string) error {
-       return fmt.Errorf(`cannot invoke snapctl operation commands (here %q) from outside of a snap`, subcommand)
+type MissingContextError struct {
+       subcommand string
+}
+
+func (e *MissingContextError) Error() string {
+       return fmt.Sprintf(`cannot invoke snapctl operation commands (here %q) from outside of a snap`, e.subcommand)
 }
 
 type baseCommand struct {
@@ -77,7 +81,7 @@ func (c *baseCommand) context() *hookstate.Context {
 
 func (c *baseCommand) ensureContext() (context *hookstate.Context, err error) {
        if c.c == nil {
-               err = createMissingContextError(c.name)
+               err = &MissingContextError{c.name}
        }
        return c.c, err
 }
diff --git a/overlord/hookstate/ctlcmd/helpers.go b/overlord/hookstate/ctlcmd/helpers.go
index 5260027224..595d99560b 100644
--- a/overlord/hookstate/ctlcmd/helpers.go
+++ b/overlord/hookstate/ctlcmd/helpers.go
@@ -131,7 +131,7 @@ func queueCommand(context *hookstate.Context, tts []*state.TaskSet) error {
 
 func runServiceCommand(context *hookstate.Context, inst *servicestate.Instruction) error {
        if context == nil {
-               return createMissingContextError(inst.Action)
+               return &MissingContextError{inst.Action}
        }
 
        st := context.State()
```
is a bit more idiomatic go. I.e. having an error type for this instead of creating different error strings. The end result is the same so it's not really a big deal but I could not *not* mention it :)

Reply:
Thanks, applied!